metals/src/main/scala/scala/meta/internal/worksheets/WorksheetProvider.scala

```diff
@@ -108,8 +110,20 @@ class WorksheetProvider(
       path: AbsolutePath,
       token: CancelToken
   ): Future[Unit] = {
-    evaluateAsync(path, token).map(
-      _.foreach(publisher.publish(languageClient, path, _))
+    val possibleBuildTarget = buildTargets.inverseSources(path)
+    val previouslyCompiled = compilations.previouslyCompiled.toSeq
+
+    (possibleBuildTarget match {
+      case Some(bdi) if previouslyCompiled.isEmpty =>
+        compilations.compileTarget(bdi).ignoreValue
```

Comment:
Could we squash the case into one with `||` ?

Reply:
Yup, fixed.